### ISSUE TITLE
Register method type hint

### ISSUE THREAD

**Issue by HamadTheIronside** [danialkeimasi/python-registerer#2]
Hey,

```python
import registerer

class Human:
    ...

gender_registery = registerer.Registerer(parent_class=Human)

@gender_registery.register("man")
class Man(Human):
    ...

print(gender_registery._registry_dict)
```

`register` method is giving me the following type:
`(method) register: (*args: Unknown, **kwargs: Unknown) -> (Type[Human] | ((item: Type[Human]) -> T@Registerer))`

And says `Expected no arguments to "Human" constructor` (Because I passed a string to register, it says this)

BTW, it gets registered, but the type hint is not correct.

Python version: 3.9.7

**Comment by danialkeimasi** [comment]
Hi,

Seems like this error only occurs with your specific language server on your IDE.
Can you provide more information about your IDE?

If you use VSCode do you have Pylance extension installed?

**Comment by danialkeimasi** [comment]
I've kinda figured out how things can go wrong. addressed this issue in 9649e099be37d776a058a9eee2817d6dffb9b3b5 and released a patch.

**Comment by danialkeimasi** [comment]
It should be fixed in v0.4.3 release.
